WordPress/src/main/java/org/wordpress/android/ui/blaze/ui/blazeoverlay/BlazeViewModel.kt

```diff
@@ -0,0 +1,83 @@
+package org.wordpress.android.ui.blaze.ui.blazeoverlay
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import dagger.hilt.android.lifecycle.HiltViewModel
+import org.wordpress.android.ui.blaze.BlazeFeatureUtils
+import org.wordpress.android.ui.blaze.BlazeFlowSource
+import org.wordpress.android.ui.blaze.BlazeUiState
+import org.wordpress.android.ui.blaze.PostUIModel
+import java.util.Locale
+import javax.inject.Inject
+
+@HiltViewModel
+class BlazeViewModel @Inject constructor(private val blazeFeatureUtils: BlazeFeatureUtils) : ViewModel() {
+    private lateinit var blazeFlowSource: BlazeFlowSource
+
+    private val _refreshAppTheme = MutableLiveData<Unit>()
+    val refreshAppTheme: LiveData<Unit> = _refreshAppTheme
+
+    private val _refreshAppLanguage = MutableLiveData<String>()
+    val refreshAppLanguage: LiveData<String> = _refreshAppLanguage
+
+    private val _uiState = MutableLiveData<BlazeUiState>()
+    val uiState: LiveData<BlazeUiState> = _uiState
+
+    private val _promoteUiState = MutableLiveData<BlazeUiState.PromoteScreen>()
+    val promoteUiState: LiveData<BlazeUiState.PromoteScreen> = _promoteUiState
+
+    fun setAppLanguage(locale: Locale) {
+        _refreshAppLanguage.value = locale.language
+    }
+
+    fun trackOverlayDisplayed() {
+        blazeFeatureUtils.trackOverlayDisplayed(blazeFlowSource)
+    }
+
+    fun onPromoteWithBlazeClicked() {
+        blazeFeatureUtils.trackPromoteWithBlazeClicked()
+    }
+
+    fun initialize(postModel: PostUIModel?) {
```

Comment:
@AjeshRPai 
🔍 Nitpick: The general pattern in the codebase has been to keep code to a single line if it fits. Same with the `showNextScreen` fun. Not asking you to change it from two lines back to one, but just curious if we (you & I) want to make this a pattern that we follow. wdyt?

Reply:
Hey @zwarm 👋🏼 

Nice catch. To be honest, It was a mistake from my part. I think we can go with the general pattern of keeping it in same line, if that's ok with you too. I have made the change to keep the code in same line, with the commit [4edceb8](https://github.com/wordpress-mobile/WordPress-Android/pull/17941/commits/4edceb8a130d31c5e68a953e5d79caa3e8d87a25).